koku/cost_models/serializers.py

```diff
@@ -157,136 +163,216 @@ def _validate_continuouse_tiers(tiers):
-            },
-            CostModelMetricsMap.OCP_METRIC_CPU_CORE_REQUEST_HOUR: {
-                'unit': 'core-hours',
-                'display_name': 'Compute request rate'
-            },
-            CostModelMetricsMap.OCP_METRIC_MEM_GB_USAGE_HOUR: {
-                'unit': 'GB-hours',
-                'display_name': 'Memory usage rate'
-            },
-            CostModelMetricsMap.OCP_METRIC_MEM_GB_REQUEST_HOUR: {
-                'unit': 'GB-hours',
-                'display_name': 'Memory request rate'
-            },
-            CostModelMetricsMap.OCP_METRIC_STORAGE_GB_USAGE_MONTH: {
-                'unit': 'GB-months',
-                'display_name': 'Volume usage rate'
-            },
-            CostModelMetricsMap.OCP_METRIC_STORAGE_GB_REQUEST_MONTH: {
-                'unit': 'GB-months',
-                'display_name': 'Volume request rate'
-            }
-        }
-        return metric_map[metric]
+        return self.metric_map[source_type][metric]
 
-    def to_representation(self, rate):
+    def to_representation(self, rate_obj):
         """Create external representation of a rate."""
-        rates = rate.rates
+        source_type = rate_obj.get('source_type')
+        metric_name = rate_obj.get('metric', {}).get('name')
+        display_data = self._get_metric_display_data(source_type, metric_name)
+        if source_type in SOURCE_TYPE_MAP:
+            source_type = SOURCE_TYPE_MAP[source_type]
 
-        provider_uuids = RateManager(rate_uuid=rate.uuid).get_provider_uuids()
-        display_data = self._get_metric_display_data(rate.metric)
         out = {
-            'uuid': rate.uuid,
-            'provider_uuids': provider_uuids,
-            'metric': {'name': rate.metric,
-                       'unit': display_data.get('unit'),
-                       'display_name': display_data.get('display_name')}
+            'metric': {
+                'name': metric_name,
+                'label_metric': display_data.label_metric,
+                'label_measurement': display_data.label_measurement,
+                'label_measurement_unit': display_data.label_measurement_unit,
+            },
+            'source_type': source_type,
         }
-        for rate_type in rates.values():
-            if isinstance(rate_type, list):
-                for rate_item in rate_type:
-                    RateSerializer._convert_to_decimal(rate_item)
-                    if not rate_item.get('usage'):
-                        rate_item['usage'] = {'usage_start': rate_item.pop('usage_start'),
-                                              'usage_end': rate_item.pop('usage_end'),
-                                              'unit': display_data.get('unit')}
+
+        # Specifically handling only tiered rates now
+        # with the expectation that this code will be generalized
+        # when other rate types (e.g. markup) are introduced
+        tiered_rates = rate_obj.get('tiered_rates', [])
```

Comment:
When everything is a tiered rate, we can simplify the name and just call it a `rate`.

Can we remove the `tiered_` bit, please?

Reply:
I'm trying to keep our existing rate model as in-tact as possible here and just focusing on translating rates to cost models. Also the intention of the comment in code here is that we are just about to handle additional "rate" types (i.e. markups) and I think we should really leave these decisions until we have additional information on those requirements. As is calling this a tiered rate is not hurting anything.

Reply:
I hope it's clear I'm objecting to the string in the data, not the variable name. I'm worried that we might be talking about different things.

I've been asking for the serializer to stop using `tiered_rate` for a while. It was specc'ed out to be changed in a previous issue, but it was not done.

I don't really understand what additional requirements are needed. The decision to call it a `tiered_rate` was arbitrary and decided by whomever wrote the serializer. Can you help me understand what additional requirements you think are needed?

Reply:
@blentz I know you like to call decisions in the work we've done arbitrary, but please stop doing that. It's not helpful and it's unfair to the person who did the previous work. Could you furnish a link to the previous issue where this change was missed? Otherwise other than your specific personal preference, I see no reason to make that change now.

Reply:
I don't attach a negative connotation to the term. I'm sorry if it sounds like I'm being negative. That's not my intent. I will try to find a better way to say it. Thank you for letting me know.

The issue where `tiered_rate` was respecc'ed to `rate` was #807  The proposed new structure was accepted, then silently not implemented without any discussion.